lib/Profile/AppointmentsAction.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * @copyright 2021 Christoph Wurst <christoph@winzerhof-wurst.at>
+ *
+ * @author 2021 Christoph Wurst <christoph@winzerhof-wurst.at>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+namespace OCA\Calendar\Profile;
+
+use OCA\Calendar\AppInfo\Application;
+use OCA\Calendar\Db\AppointmentConfig;
+use OCA\Calendar\Service\Appointments\AppointmentConfigService;
+use OCP\IL10N;
+use OCP\IURLGenerator;
+use OCP\IUser;
+use OCP\IUserSession;
+use OCP\Profile\ILinkAction;
+use function array_filter;
+use function count;
+
+class AppointmentsAction implements ILinkAction {
+
+	/** @var IL10N */
+	private $l10n;
+
+	/** @var IUserSession */
+	private $userSession;
+
+	/** @var AppointmentConfigService */
+	private $configService;
+
+	/** @var IURLGenerator */
+	private $urlGenerator;
+
+	/** @var IUser|null */
+	private $targetUser;
+
+	/** @var AppointmentConfig[] */
+	private $configs = [];
+
+	public function __construct(IL10N $l10n,
+								IUserSession $userSession,
+								AppointmentConfigService $configService,
+								IURLGenerator $urlGenerator) {
+		$this->l10n = $l10n;
+		$this->userSession = $userSession;
+		$this->configService = $configService;
+		$this->urlGenerator = $urlGenerator;
+	}
+
+	public function preload(IUser $targetUser): void {
+		$this->targetUser = $targetUser;
+
+		$this->configs = array_filter(
```

Comment:
Instead of filtering here, let's add the possibility in `AppointmentConfigMapper` to filter at the database level directly, and add an appropriate method in `AppointmentConfigService`.